### PR TITLE
Augment Course to store corequisites of the course from the API data 

### PR DESCRIPTION
**Pull request by lorena-b**
# Description

Modified the Course class so that can store course corequisites, with the same logic used to store exclusions. 

## Type of change

- [x] New feature 

# How Has This Been Tested?

-Manual testing ex: MAT240 and seeing the returned coreq (MAT157)
-Added tests to course creator to confirm the coreqs are stored properly from the API data 

# Does this PR follow the rubric?

No significant SOLID/design pattern usage 

# Checklist:

- [x] I ran `./gradlew spotlessApply`
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I ran `./gradlew build` and the gradle build succeeded
- [x] I have not violated any design principles

### CODE REVIEW

File: src/main/java/entities/Course.java [hunk 1]
```diff
@@ -24,12 +26,14 @@ public Course(
             ArrayList<Section> lectures,
             ArrayList<Section> tutorials,
             char session,
-            ArrayList<String> exclusions) {
+            ArrayList<String> exclusions,
```
Comment (sugolov) [inline src/main/java/entities/Course.java:29]: We should stick to Liskov substitution principle and code to interfaces. I'd change these to List

File: src/main/java/entities/Course.java [hunk 2]
```diff
@@ -48,6 +52,10 @@ public String getCourseId() {
         return courseExclusionIds;
     }
 
+    public ArrayList<String> getCorequisites() {
```
Comment (sugolov) [inline src/main/java/entities/Course.java:55]: list

File: src/main/java/workers/CourseCreator.java [hunk 3]
```diff
@@ -88,23 +93,50 @@ public static Course generateCourse(String courseId, char session) throws IOExce
...
     }
 
+    /**
+     * Extracts all course names from a string containing the course corequisites and puts it into
+     * an ArrayList
+     *
+     * @param value a String that corresponds to all the corequisites for a course
+     * @return an ArrayList of course names
+     */
+    public static ArrayList<String> getCourseCorequisites(String value) {
```
Comment (sugolov) [inline src/main/java/workers/CourseCreator.java:116]: list

**Review by sugolov — COMMENTED** [review]
Overall good, remember to code to interfaces and not implementations (List vs. ArrayList). We only refactored this recently elsewhere in the code, but I'd add this in here to be consistent.

File: src/main/java/entities/Course.java (hunk 1 above)
Comment (lorena-b) [reply src/main/java/entities/Course.java:29]: Yes Sir! :D

File: src/main/java/workers/CourseCreator.java [hunk 4]
```diff
@@ -88,23 +93,50 @@ public static Course generateCourse(String courseId, char session) throws IOExce
...
+
+    /**
+     * Helper method for extracting the course code values from the API contents
+     *
+     * @param value a String that corresponds to all the corequisites for a course
+     * @param shortenedCodes an empty arraylist to add the course codes to
+     */
+    private static void extractCodes(String value, ArrayList<String> shortenedCodes) {
+        String cleanedValue = value.replace("\"", "").replace(".", "");
+        ArrayList<String> values = new ArrayList<>(List.of(cleanedValue.split("\\s*,\\s*")));
```
Comment (kenneth-miura) [inline src/main/java/workers/CourseCreator.java:134]: if u make values a generic List u can just do `List<String> values = List.of(cleanedValue.split("\\s, \\s*"))`

File: src/test/java/workersTests/CourseCreatorTest.java [hunk 5]
```diff
@@ -56,6 +55,20 @@ public void testMockExclusion2() {
         assertEquals(expected, mockCourse2.getExclusions());
     }
 
+    @Test(timeout = 1000)
+    public void testMockCoreq1() {
+        ArrayList<String> expected = new ArrayList<>(List.of("TST103"));
```
Comment (kenneth-miura) [inline src/test/java/workersTests/CourseCreatorTest.java:60]: if u make expected a List u can directly use `List.of`

**Review by kenneth-miura — CHANGES_REQUESTED** [review]
sorry did not get notif about PR review request. Looks good generally, i agree with what anton put. I commented on a few lines where u can simplify by just using generic List

**Pull request merged**